sormas-ui/src/main/java/de/symeda/sormas/ui/labmessage/LabMessageController.java

```diff
@@ -898,4 +923,32 @@ private HorizontalLayout getLabMessageButtonsPanel(LabMessageDto labMessage, Run
 
 		return buttonsPanel;
 	}
+
+	public Optional<byte[]> convertToPDF(String labMessageUuid) {
+
+		LabMessageDto labMessageDto = FacadeProvider.getLabMessageFacade().getByUuid(labMessageUuid);
+
+		try {
+			ExternalMessageResult<byte[]> result = FacadeProvider.getExternalLabResultsFacade().convertToPDF(labMessageDto);
+
+			if (result.isSuccess()) {
+				return Optional.of(result.getValue());
+			} else {
+				new Notification(
+						I18nProperties.getString(Strings.headingLabMessageDownload),
+						I18nProperties.getString(Strings.messageLabMessageDownloadConvertFailed),
+						Notification.Type.ERROR_MESSAGE,
+						false).show(Page.getCurrent());
+			}
+
+		} catch (NamingException e) {
+			new Notification(
+					I18nProperties.getString(Strings.headingLabMessageDownload),
+					I18nProperties.getString(Strings.messageLabMessageDownloadConvertFailed),
```

Comment:
Mostly, the error messages are directly transmitted from the adapter. Why did you decide to store this message in SORMAS?